src/main/java/frc/trigon/robot/commands/commandfactories/CollectionCommands.java

```diff
@@ -0,0 +1,57 @@
+package frc.trigon.robot.commands.commandfactories;
+
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.trigon.robot.RobotContainer;
+import frc.trigon.robot.commands.CommandConstants;
+import frc.trigon.robot.commands.commandclasses.CoralAlignmentCommand;
+import frc.trigon.robot.subsystems.coralintake.CoralIntakeCommands;
+import frc.trigon.robot.subsystems.coralintake.CoralIntakeConstants;
+import org.trigon.hardware.misc.leds.LEDCommands;
+
+/**
+ * A class that contains commands for collecting game pieces.
+ */
+public class CollectionCommands {
+    public static boolean SHOULD_ALIGN_TO_CORAL = true;
+
+    public static Command getCoralCollectionCommand() {
+        return getInitiateCoralCollectionCommand().unless(RobotContainer.CORAL_INTAKE::hasGamePiece).alongWith(CommandConstants.COLLECTION_RUMBLE_COMMAND);
+    }
+
+    private static Command getInitiateCoralCollectionCommand() {
+        return new ParallelCommandGroup(
+                new CoralAlignmentCommand().onlyIf(() -> SHOULD_ALIGN_TO_CORAL),
+                LEDCommands.getBlinkingCommand(Color.kAqua, CoralIntakeConstants.COLLECTION_LEDS_BLINKING_SPEED).unless(() -> SHOULD_ALIGN_TO_CORAL),
+                CoralIntakeCommands.getSetTargetStateCommand(CoralIntakeConstants.CoralIntakeState.COLLECT)
+        ).until(() -> RobotContainer.CORAL_INTAKE.isEarlyCoralCollectionDetected() || RobotContainer.CORAL_INTAKE.hasGamePiece())
+                .finallyDo((interrupted) -> {
+                    if (!interrupted)
+                        getLoadCoralCommand().schedule();
+                });
+    }
+
+    private static Command getLoadCoralCommand() {
```

Comment:
you don't want any other elevaor/intake/gripper commands to run while this happens